Userland/Services/WebDriver/Session.cpp

```diff
@@ -106,4 +114,55 @@ Web::WebDriver::Response Session::stop()
     return JsonValue {};
 }
 
+// 11.2 Close Window, https://w3c.github.io/webdriver/#dfn-close-window
+Web::WebDriver::Response Session::close_window()
+{
+    // 3. Close the current top-level browsing context.
+    TRY(web_content_connection().close_window());
+    m_windows.remove(m_current_window_handle);
+
+    // 4. If there are no more open top-level browsing contexts, then close the session.
+    if (m_windows.is_empty())
+        stop();
+
+    // 5. Return the result of running the remote end steps for the Get Window Handles command.
+    return get_window_handles();
+}
+
+// 11.3 Switch to Window, https://w3c.github.io/webdriver/#dfn-switch-to-window
+Web::WebDriver::Response Session::switch_to_window(StringView const& handle)
+{
+    // 4. If handle is equal to the associated window handle for some top-level browsing context in the
+    //    current session, let context be the that browsing context, and set the current top-level
+    //    browsing context with context.
+    //    Otherwise, return error with error code no such window.
+    auto const& maybe_window = m_windows.get(handle);
+
+    if (!maybe_window.has_value())
+        return Web::WebDriver::Error::from_code(Web::WebDriver::ErrorCode::NoSuchWindow, "Window not found");
+
+    m_current_window_handle = String::from_utf8(handle).release_value_but_fixme_should_propagate_errors();
```

Comment:
On closer look, I think we can avoid this allocation altogether and just copy the key's String, by using `find` instead of `get`. So something like:
```c++
    if (auto it = m_windows.get(handle); it != m_windows.end())
        m_current_window_handle = it->key;
    else
        return Web::WebDriver::Error::from_code(Web::WebDriver::ErrorCode::NoSuchWindow, "Window not found");
```

Reply:
got it and applied suggested change. thank you.